### PR TITLE
React Book Store: Milestone 2

### PR DESCRIPTION
**Pull request by mahmoud717**
In this milestone I:
- added the necessary structure for the app.
- initiated redux store and implemented across the app.
- added forms for new books, editing books, update progress status. I also added the corresponding actions and components.
- finished the app. 
- published the app again to netlify [here](https://www.book-store-cms.netlify.app/).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-  <React.StrictMode>
+const store = createStore(rootReducer);
 
-    <App />
+export default store;
 
+ReactDOM.render(
+  <React.StrictMode>
+    <Provider store={store}><App /></Provider>
   </React.StrictMode>,
```
Comment (whiz25) [inline src/index.js:17]: - Add example books to initial state.
```javascript
const INITIAL_STATE = {
  books: [
    { title: "Romeo and Juliet" },
    { title: "How to cook your first tomato soup - Practical guide" },
    { title: "Everything you need to know about startups" }
  ]
}

const store = createStore(rootReducer, INITIAL_STATE);

```

File: src/components/booklist.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        tempData[index] = state.Books[index];
+      }
+    }
+    books = tempData;
+  }
+  return ({ books });
+};
+export default connect(
+  mapStateToProps, null,
+)(booklist);
```
Comment (whiz25) [inline src/components/booklist.js]: - Create a new folder called `containers` and move this file there. It holds a list of books more like a container.
- Render the books in a table
```html
<table>
     <thead>
            <tr>
              <th>ID</th>
              <th>Title</th>
              <th>Category</th>
            </tr>
          </thead>
          <tbody>
            {books}
          </tbody>
  </table>
```

File: src/actions/BookActions.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const UPDATE_PROGRESS = 'UPDATE_PROGRESS';
+
+export const addBook = book => ({
+  type: ADD_BOOK,
+  payload: book,
+});
+
+export const removeBook = bookID => ({
+  type: REMOVE_BOOK,
+  payload: bookID,
```
Comment (whiz25) [inline src/actions/BookActions.js:12]: - Take a book object as a parameter. You can then access the ID from the object itself. 
- Move all these action objects to `src/actions/index.js` file.

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray:
Nice work on the milestone. You have implemented most of the features required for this PR.
However, we need to make some changes before we can merge this PR.
Check my comments for suggestions.


### Martin Afani

File: src/actions/BookActions.js (hunk 3 above)
Comment (mahmoud717) [reply src/actions/BookActions.js:12]: - this is required in the book component not here.
- all actions are moved.

File: src/components/booklist.js (hunk 2 above)
Comment (mahmoud717) [reply src/components/booklist.js]: I moved the booklist file to the container folder.
as for the table, it's not a project specific requirement and it's going to be deleted in the next milestone.
if you take a look at the live demo you'll find that I made all of the requirements for all 5 milestones.

File: src/index.js (hunk 1 above)
Comment (mahmoud717) [reply src/index.js:17]: it's not a project specific requirement and it's going to be deleted in the next milestone.
if you take a look at the live demo you'll find that I made all of the requirements for all 5 milestones.

**Comment by mahmoud717** [comment]
Hey there, for this project I forgot to request a CRR for each milestone and the project is complete.
requirements for all five milestones are added and I added some more features such as editBook and updateProgress.
If you are going to review this project, please don't ask for changes that are going to be undone in the next milestone.
I apologize for my fault And I hope you accept the project.

File: src/index.js (hunk 1 above)
Comment (mahmoud717) [reply src/index.js:17]: there are book examples add to the reducers/books.js file.

File: src/components/booklist.js (hunk 2 above)
Comment (bellom) [reply src/components/booklist.js]: @mahmoud717

You should also move `bookform` to the container folder

React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory, (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @mahmoud717   :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

It's a little bit weird that you forgot to create PR for each milestone because each milestone on the pathwright is on a different page.

Do well to carefully read the pathwright instructions before jumping to code, this would be helpful for you in the future.

# Status: Changes Required! ♻️ 

**Kindly check out my comments above;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by mahmoud717** [comment]
Hey there, all changes are made. the only change that I didn't make is to add the book as a table because it's going to be removed in the next milestone. and as I explained, the project is complete.

File: src/components/booklist.js (hunk 2 above)
Comment (bellom) [reply src/components/booklist.js]: @mahmoud717, can you please take a good look at my comment.

**Comment by bellom** [comment]
> Hey there, all changes are made. the only change that I didn't make is to add the book as a table because it's going to be removed in the next milestone. and as I explained, the project is complete.

@mahmoud717 I Agreed with you but kindly carefully read my comment above, in fact, I mentioned you so you should get an email alert about it, I'm not talking about adding the book as a table.

I'm talking about the file structure which I don't think would be deprecated in the next milestone, kindly correct me if I'm wrong.

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @mahmoud717  :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**